src/examples/Makefile

```diff
@@ -3,38 +3,40 @@
 #
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
 
+include ../config/Makefile.config
```

Comment:
Hi @liruipeng .  We intentionally avoid coupling the examples to the hypre configuration stuff to make them as simple and instructive as possible.  Can you do this in a way that is more in the spirit of the original Makefile?

Reply:
Hi, @rfalgout For GPU build, we need to change `CC` and more others. What is a more preferred way of doing it? Thanks!

Reply:
Good question.  The main reason for the examples is to teach people how to use hypre, so we need to keep them simple.  Are you updating this so you can run the examples on GPUs as part of regression testing?  I'm not sure that would be a good reason, since we can test the code through our drivers.  If you want to illustrate how to use GPUs, then it might be better to introduce a "GPU example" that illustrates how the user code needs to change.  I would like to hear other opinions on this, though.

Reply:
@rfalgout Thanks for the suggestions. I changed `Makefile` back, and added `Makefile_gpu`. So, one can compile the examples with `make -f Makefile_gpu` if hypre has been built with GPUs. Looks better now?

I actually want to have the sample examples as for CPUs with minor changes as you may have seen in the 'malloc'. So, these examples should work on GPUs as well, not performing the best but should be good enough to show the usage. Some of them don't have GPU-support yet for different reasons.

Any thoughts? Thanks!

Reply:
Hi @liruipeng .  Yes, this looks much better.  Thanks!